### DISCUSSION TITLE
Run custom command by init?

### DISCUSSION THREAD

**Discussion by sotiris-bos** [zbm-dev/zfsbootmenu#565] [Questions]
Hello, I am curious if I can modify the zfsbootmenu start procedure to run a custom command before the encryption key load screen.

Specifically, I want to include an mqtt client in the initramfs and a script that, after the network is online, publishes to an mqtt topic so that home assistant will turn on my TV, which I am using as a monitor, so I don't have to turn it on by myself.

This was easy with dracut and systemd when I was using GRUB but I have no idea where to start applying my changes here.

Any pointers to what files I need to edit?

**Comment by zdykstra** [comment]
I can't tell if you're joking or not - but the [hook system](https://docs.zfsbootmenu.org/en/v2.3.x/man/zfsbootmenu.7.html#user-hooks) is designed for this soft of insanity.
